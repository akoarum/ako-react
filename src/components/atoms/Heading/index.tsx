import styled from 'styled-components'
import * as React from 'react'
import * as style from '../../../utils/style'

type Props = {
  className?: string
  level?: number
  visual?: number
}

const StyledHeading = styled.div`
  .-lv2 {
    ${style.mediaQuery.pc} {
      font-size: 2.3rem;
      line-height: 1.4;
    }
    ${style.mediaQuery.sp} {
      font-size: 2.2rem;
      line-height: 1.5;
    }
  }
  .-lv3 {
    font-size: 2rem;
    line-height: 1.5;
  }
  .-lv4 {
    font-size: 1.8rem;
    line-height: 1.5;
  }
`

export const Heading: React.FC<Props> = ({
  className,
  level = 2,
  visual = 2,
  children
}) => {
  const Tag: any = `h${level}`
  const classNames: string[] = []

  if (className) {
    classNames.push(className)
  }

  classNames.push(`-lv${visual}`)

  return (
    <StyledHeading>
      <Tag className={classNames.join(' ')}>{children}</Tag>
    </StyledHeading>
  )
}

export default Heading
