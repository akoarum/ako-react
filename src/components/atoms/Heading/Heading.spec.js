import React from 'react'
import renderer from 'react-test-renderer'
import Heading from './index'

describe('Heading', () => {
  let wrapper

  beforeEach(() => {
    wrapper = renderer.create(
      <Heading>テキスト</Heading>
    )
  })

  describe('initialize', () => {
    it('snapshot', () => {
      const tree = wrapper.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
