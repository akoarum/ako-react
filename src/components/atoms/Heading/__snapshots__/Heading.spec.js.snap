// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading initialize snapshot 1`] = `
<div
  className="sc-bdVaJa kRmqUI"
>
  <h2
    className="-lv2"
  >
    テキスト
  </h2>
</div>
`;
